### ISSUE TITLE
prevent forking unsaved shader

### ISSUE THREAD

**Issue by FabriceNeyret** [patuwwy/ShaderToy-Chrome-Plugin#59]
I lost 2 shaders yesterday. I'm not 100% sure of what I did (several open tabs...), but one possibility is:
If I modify a shader then fork, the modifications are lost (this can happen by confusing "fork" with "save as new", or by miss-clicking "fork" instead of "save" after a first forking, or by just forgetting to save modifications in the original. User's errors, but not unlikely.)

-> if the shader is unsaved, could you ask confirmation in a requester before forking ?
( the message could also suggest to "either save first then fork, or to save as new instead of fork").

thanks !

**Comment by patuwwy** [comment]
I agree, the message must be added if any changes are not saved while editing your own shader.
"Fork" takes current shader contents.

So You can lost modifications in following scenario:

1. Fork someone's shader. (A → B)
2. Modify it (B → B'), 
3. Fork ( B' → C ) 
4. Modify (C → C')

B is lost.

**Comment by FabriceNeyret** [comment]
- "  'Fork' takes current shader contents. ": 
NB: yes for the code, but nothing else (title, description, tags).

- Scenario for loosing a shader: 
I don't know if it's what happened in my case, but closing this possibility can only be good :-)
